test/resources/tzbtc-parameter-entrypoints-ref.mtz

```diff
@@ -3,8 +3,8 @@
 
 parameter (or (or (or (pair %getVersion (unit %viewParam)
                                         (contract %viewCallbackTo nat))
-                      (pair %getAllowance (pair %viewParam (address %owner)
-                                                           (address %spender))
+                      (pair %getAllowance (pair %viewParam (address :owner)
+                                                           (address :spender))
```

Comment:
Interesting. According to TZIP-7 these should be type annotations, so the new version is right. How did it happen that the previous version was wrong and how did you spot it?

Reply:
I dunno actually, it just happened automatically during Morley update. Probably this repository was so outdated, that the earlier made `% -> :` change in Morley reached here only now.

Reply:
> the earlier made `% -> :` change in Morley

But did we ever make such a chance? IIRC we've been always using `Named` for these things and `Named` was always translated to type annotations.

Reply:
Hm, really. Not sure how did it work wrong before then

Reply:
`getAllowance` annotations have been changed in b0aa3a80f9ca74513a1fb66e515a2f580099bbde. I don't understand how it could happen there, that commit affects only metadata, it doesn't update any revisions. But I suppose this change is just misplaced, probably you regenerated that file just once, but it was changed in more then 1 commit.

Reply:
In `morley-ledgers` revision `7f0cbca` (the one used here) there are field annotations (which is wrong). In latest master (which uses newer `morley` and `lorentz` apparently) annotations are correct. I will continue investigation.

Reply:
Okay, I am such a :male_detective: :clap: 
At some point the following very hacky instance was added to `morley-ledgers`:
```
-- The following instances cause the field annotations to appear insted of type annotations
-- as it would be the case by default.
instance {-# OVERLAPPING #-}
  (KnownSymbol s1, HasAnnotation a, HasAnnotation b) => HasAnnotation (s1 :! a, b) where
    getAnnotation fe = NTPair noAnn (ann $ symbolValT' @s1) noAnn (getAnnotation @a fe) (getAnnotation @b fe)
```

It was added for FA2, but accidentally affected FA1.2 stuff as well. It was removed in https://gitlab.com/morley-framework/morley-ledgers/-/merge_requests/15/diffs which reached `tezos-btc` in this current PR. So ok, now we know why this change happened. I should also create a follow-up issue to make sure that the check for FA1.2 compliance actually works (I mean parameter type check).

Reply:
[Here](https://gitlab.com/morley-framework/morley-ledgers/-/issues/15)

Reply:
Oh cool, nice investigation :clap: 

Hopefully, people will learn to avoid overlapping instances in inappropriate places and use newtypes instead of hacks.

Reply:
Created this [notion page](https://www.notion.so/serokell/Using-overlapping-instances-cea84f821adc49bab570d93f078b8858).

Reply:
IIRC those instances were first created as a quick hack in `stablecoin` (which was not supposed to be used as a library) and then were moved to `morley-ledgers`.